File: app/src/main/java/com/nativo/nativo_android_unifiedsample/NativeAdImpl/NativeAd.java
package com.nativo.nativo_android_unifiedsample.NativeAdImpl;

import android.content.Context;
import android.graphics.Color;
import android.support.v7.widget.CardView;
import android.view.View;
import android.widget.ImageView;
import android.widget.LinearLayout;
import android.widget.TextView;

import com.nativo.nativo_android_unifiedsample.R;

import net.nativo.sdk.ntvadtype.nativead.NtvNativeAdInterface;

import java.text.SimpleDateFormat;
import java.util.Date;
import java.util.Locale;

public class NativeAd implements NtvNativeAdInterface {

    private LinearLayout layout;
    private CardView cardView;
    private TextView titleLabel;
    private TextView authorLabel;
    private ImageView image;
    private ImageView sponsoredIndicator;
    private TextView sponsoredTag;
    private View view;

    @Override
    public TextView getTitleLabel() {
        if (titleLabel == null) {
            titleLabel = (TextView) view.findViewById(R.id.article_title);
            return titleLabel;
        }
        return titleLabel;
    }

    @Override
    public TextView getAuthorLabel() {
        return authorLabel;
    }

    @Override
    public TextView getPreviewTextLabel() {
        return null;
    }

    @Override
    public ImageView getPreviewImageView() {
        return image;
    }

    @Override
    public ImageView getAuthorImageView() {
        return null;
    }

    @Override
    public TextView getDateLabel() {
        return null;
    }

    @Override
    public void displaySponsoredIndicators(boolean b) {
        if (cardView != null) {
            cardView.setBackgroundColor(Color.LTGRAY);
        } else if (view != null){
            view.setBackgroundColor(Color.LTGRAY);
        }
        sponsoredIndicator.setVisibility(View.VISIBLE);
        sponsoredTag.setVisibility(View.VISIBLE);
    }

    @Override
    public String formatDate(Date date) {
        return new SimpleDateFormat("dd/MM/yyyy", Locale.US).format(date);
    }

    @Override
    public int getLayout(Context context) {
        return R.layout.article;
    }

    @Override
    public void bindViews(View v) {
        view = v;
        layout = v.findViewById(R.id.article_layout);
        cardView = v.findViewById(R.id.article_constraint_layout);
        titleLabel = v.findViewById(R.id.article_title);
        authorLabel = v.findViewById(R.id.article_author);
        image = v.findViewById(R.id.article_image);
        sponsoredIndicator = v.findViewById(R.id.video_sponsored_indicator);
        sponsoredTag = v.findViewById(R.id.sponsored_tag);
    }
}

File: app/src/main/java/com/nativo/sampleapp/NativeAdImpl/NativeAd.kt
package com.nativo.sampleapp.NativeAdImpl

import android.content.Context
import android.graphics.Color
import android.view.View
import android.widget.FrameLayout
import android.widget.ImageView
import android.widget.TextView
import androidx.cardview.widget.CardView
import com.nativo.sampleapp.R
import net.nativo.sdk.injectable.NtvNativeAdInjectable
import java.text.SimpleDateFormat
import java.util.Date
import java.util.Locale

class NativeAd : NtvNativeAdInjectable {

    private val cardView: CardView? = null
    override lateinit var titleLabel: TextView
    override lateinit var authorLabel: TextView
    override lateinit var authorImageView: ImageView
    override lateinit var previewImageView: ImageView
    override var isiContentView: FrameLayout? = null
    override val previewTextLabel: TextView? = null
    override val dateLabel: TextView? = null
    override val adChoicesImageView: ImageView? = null
    private val sponsoredTag: TextView? = null
    override lateinit var view: View

    override fun displaySponsoredIndicators(b: Boolean) {
        cardView?.setBackgroundColor(Color.LTGRAY)
        if (sponsoredTag != null) {
            sponsoredTag.visibility = View.VISIBLE
        }
    }

    override fun formatDate(date: Date?): String? {
        return SimpleDateFormat("dd/MM/yyyy", Locale.US).format(date)
    }

    override fun getLayout(context: Context): Int {
        return R.layout.native_article_2
    }

    override fun bindViews(v: View) {
        view = v
        titleLabel = v.findViewById(R.id.article_title)
        authorLabel = v.findViewById(R.id.article_author)
        previewImageView = v.findViewById(R.id.article_image)
        authorImageView = v.findViewById(R.id.article_author_image)
//        isiContentView = v.findViewById(R.id.isi_content)

//        layout = v.findViewById(R.id.article_layout);
//        adChoicesIndicator = v.findViewById(R.id.adchoices_indicator);
//        sponsoredTag = v.findViewById(R.id.sponsored_tag);
//        articleDateLabel = v.findViewById(R.id.article_date);
//        articlePreviewLabel = v.findViewById(R.id.article_description);
    }

    override val shouldPrependAuthorByline: Boolean
        get() = true
}